apps/web/components/dashboard/bookmarks/EditorCard.tsx

```diff
@@ -181,6 +188,20 @@ export default function EditorCard({ className }: { className?: string }) {
                 }
                 handlePaste(e);
               }}
+              onInput={(e: React.FormEvent<HTMLTextAreaElement>) => {
+                // Expand the textarea to a max of half the screen size in the list layout only
+                if (bookmarkLayout === "list") {
+                  const target = e.target as HTMLTextAreaElement;
+                  const maxHeight = window.innerHeight * 0.5;
+                  target.style.height = "auto";
+
+                  if (target.scrollHeight <= maxHeight) {
+                    target.style.height = `${target.scrollHeight}px`;
+                  } else {
+                    target.style.height = `${maxHeight}px`;
+                  }
+                }
+              }}
```

Comment:
Can be extracted as a separate method `handleInput`, instead of keeping it in the template